libuavcan/include/uavcan/build_config.hpp

```diff
@@ -220,11 +221,20 @@ namespace uavcan
 static const unsigned MemPoolBlockSize = UAVCAN_MEM_POOL_BLOCK_SIZE;
 #elif defined(__BIGGEST_ALIGNMENT__) && (__BIGGEST_ALIGNMENT__ <= 8)
 /// Convenient default for GCC-like compilers - if alignment allows, pool block size can be safely reduced.
+#ifdef UAVCAN_USE_FD
```

Comment:
Should be `#if defined(UAVCAN_USE_FD) and UAVCAN_USE_FD` (MISRA 16-0-7). This also applies to all other usages of this and other preprocessor macros for conditional compilation.

Reply:
Yep. Will do. I'll take a look at the Zubax conventions but it'd really help if you had a clang-format file?

Reply:
I know but we don't have it. As I just wrote elsewhere, we have a config for Uncrustify, but that tool tends to break formatting occasionally, so YMMV.

Reply:
I'm a bit confused now that I'm looking at this change. Everything else in this file is using #ifdef. Does this mean you just have a lot of MISRA cleanup or are you ignoring this rule in your coverity scans?

Reply:
This applies only to boolean values because a symbol can be explicitly defined as 0 in order to disable some feature; with `#ifdef` this will lead to errors. E.g.:

```c++
#define USE_FEATURE 0
// ...
#ifdef USE_FEATURE
// bug
#endif
```

This also does not apply if we are defining a default value:

```c++
#ifdef USE_FEATURE
# define USE_FEATURE 0
#endif
```

I skimmed though that file again and couldn't find any violations manually.